packages/eslint-plugin/src/rules/space-infix-ops.ts

```diff
@@ -132,13 +168,44 @@ export default util.createRule<Options, MessageIds>({
       );
 
       checkAndReportAssignmentSpace(node, leftNode, rightNode);
+      if (
+        node.typeAnnotation.type === AST_NODE_TYPES.TSUnionType ||
+        node.typeAnnotation.type === AST_NODE_TYPES.TSIntersectionType
+      ) {
+        checkForTypeAnnotationSpace(node.typeAnnotation);
+      }
+    }
+
+    /**
+     * Check if it has an assignment char and report if it's faulty
+     * @param node The node to report
+     */
+    function checkForIntercaceDeclarationSpace(
+      node: TSESTree.TSInterfaceDeclaration,
+    ): void {
+      const properties = node.body.body;
+
+      properties.forEach((prop: TSESTree.TypeElement) => {
+        if (prop.type === AST_NODE_TYPES.TSPropertySignature) {
+          const propTypeAnnotation = prop.typeAnnotation!;
+
+          const typeAnnotation = propTypeAnnotation.typeAnnotation;
+          if (
+            typeAnnotation.type === AST_NODE_TYPES.TSUnionType ||
+            typeAnnotation.type === AST_NODE_TYPES.TSIntersectionType
+          ) {
+            checkForTypeAnnotationSpace(typeAnnotation);
+          }
+        }
+      });
     }
 
     return {
       ...rules,
       TSEnumMember: checkForEnumAssignmentSpace,
       ClassProperty: checkForClassPropertyAssignmentSpace,
-      TSTypeAliasDeclaration: checkForTypeAliasAssignmentSpace,
+      TSTypeAliasDeclaration: checkForTypeAliasAssignmentAndTypeAnnotationSpace,
+      TSInterfaceDeclaration: checkForIntercaceDeclarationSpace,
```

Comment:
I think you could do this simpler by just using a selector on the union/intersection nodes. That would let you handle more cases like return types, variable annotations and generic types.

```ts
      TSUnionType: checkForTypeAnnotationSpace,
      TSIntersectionType: checkForTypeAnnotationSpace,
``

Reply:
Soooo much more simplier. Was able to remove a lot of code. Took time to also change isSpaceBetweenTokens for isSpaceBetween since it was deprecated. I took the liberty to refactor the other functions because i had difficulty reading it